File: 125-valid-palindrome/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        con=''
        for c in s:
            if c.isalnum():
                con +=c.lower()
        l,r = 0, len(con)-1
        while l<r:
            if con[l] != con[r]:
                return False
            l+=1
            r-=1
        return True

File: 125-valid-palindrome/125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int l = 0;
        int r = s.length()-1;
        
        while (l<r){
            Character start = s.charAt(l);
            Character end = s.charAt(r);
            
            if (!Character.isLetterOrDigit(start)){
                l++;
            }
            else if(!Character.isLetterOrDigit(end)){
                r--;
            }
            else if (Character.toLowerCase(start) != Character.toLowerCase(end)){
                return false;
            }
            else{
            l++;
            r--;}
            
        }
        return true;
    }
}